filters.py

```diff
@@ -0,0 +1,46 @@
+from types import UnicodeType, StringType
+import re
+
+class FilteringFailedException(Exception):
+    pass
+
+class BasicFilterer(object):
+    """Filter that reads in a dictionary and filters based on feature properties"""
```

Comment:
You call it a Filterer in the classname, but a filter here. Filter seems better :)